src/unit_tests/remoted/test_sendmsg.c

```diff
@@ -164,10 +298,313 @@ void test_send_msg_tcp_err(void ** state) {
     assert_int_equal(remoted_state.queued_msgs, 0);
 }
 
+void test_send_msg_tcp_err_closed_socket(void ** state) {
+    (void) state;
+
+    char *agent_id = "001";
+    char *msg = "abcdefghijk";
+    ssize_t msg_length = 11;
+    int key = 0;
+
+    char *crypto_msg = "!@#123abc";
+    ssize_t crypto_size = 9;
+
+    logr.global.agents_disconnection_time = 0;
+    remoted_state.queued_msgs = 0;
+
+    // Setup closed socket
+    keys.keyentries[0]->sock=-1;
+
+    expect_function_call(__wrap_pthread_rwlock_rdlock);
+
+    expect_string(__wrap_OS_IsAllowedID, id, agent_id);
+    will_return(__wrap_OS_IsAllowedID, key);
+
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    will_return(__wrap_time, (time_t)0);
+
+    expect_function_call(__wrap_pthread_mutex_unlock);
+
+    expect_string(__wrap_CreateSecMSG, msg, msg);
+    expect_value(__wrap_CreateSecMSG, msg_length, msg_length);
+    expect_value(__wrap_CreateSecMSG, id, key);
+    will_return(__wrap_CreateSecMSG, crypto_size);
+    will_return(__wrap_CreateSecMSG, crypto_msg);
+
+    expect_function_call(__wrap_pthread_mutex_lock);
+
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    expect_function_call(__wrap_pthread_rwlock_unlock);
+
+    expect_string(__wrap__mdebug1, formatted_msg, "Send operation cancelled due to closed socket.");
+
+    int ret = send_msg(agent_id, msg, msg_length);
+
+    assert_int_equal(ret, -1);
+    assert_int_equal(remoted_state.queued_msgs, 0);
+}
+
+void test_send_msg_udp_ok(void ** state) {
+    (void) state;
+
+    char *agent_id = "001";
+    char *msg = "abcdefghijk";
+    ssize_t msg_length = 11;
+    int key = 0;
+
+    char *crypto_msg = "!@#123abc";
+    ssize_t crypto_size = 9;
```

Comment:
Hi @jftuduri

Same change here

```suggestion
    ssize_t crypto_size = strlen(msg);
```